### PR TITLE
chore: changelog for 0.10.0

### PR DESCRIPTION
**Pull request by stevenpyzhang**
### Description 
Preliminary changelog for 0.10.0 release

### Testing done 
_Describe the testing strategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,136 @@
...
+* Java client error handling ([#5360](https://github.com/confluentinc/ksql/pull/5360)) ([4dca4dc](https://github.com/confluentinc/ksql/commit/4dca4dc4f56d9448bc0e66ea39cc728b2cf0a8fa))
+* KSQL does not accept more queries when running QueryLimit - 1 queries ([#5461](https://github.com/confluentinc/ksql/pull/5461)) ([d64f1bc](https://github.com/confluentinc/ksql/commit/d64f1bc7ba4359b3d7a8cf3ba28d315eef47389e))
+* make stream and column names case-insensitive in /inserts-stream ([#5591](https://github.com/confluentinc/ksql/pull/5591)) ([e9e3042](https://github.com/confluentinc/ksql/commit/e9e3042f1bdbd6da384146410f866aab6b12b987))
+* NPE in latest_by_offset if first element being processed has null… ([#4975](https://github.com/confluentinc/ksql/pull/4975)) ([a9668d2](https://github.com/confluentinc/ksql/commit/a9668d245db797e193d1e4bcc8734eab29ba99b3))
+* Override Scala version in our build ([#5381](https://github.com/confluentinc/ksql/pull/5381)) ([7cf82e4](https://github.com/confluentinc/ksql/commit/7cf82e443283bbd189944a15a8341646919cf01f))
```
Comment (big-andy-coates) [inline CHANGELOG.md:57]: remove: Internal detail

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,136 @@
...
+* /query-stream endpoint should serialize Struct (MINOR) ([#5205](https://github.com/confluentinc/ksql/pull/5205)) ([12b092b](https://github.com/confluentinc/ksql/commit/12b092bda160f614ef0b8096f77b7013efc3b190))
+* Move Cors handler in front of /chc handlers ([#5239](https://github.com/confluentinc/ksql/pull/5239)) ([004ced2](https://github.com/confluentinc/ksql/commit/004ced255479f6a5975848270746a45746a4437a))
+* use schema in annotation as schema provider if present ([1a90eeb](https://github.com/confluentinc/ksql/commit/1a90eeb71ae8292410fc99d49c24fdafadcf9de7))
+* use sr's jackson-jsonschema version ([#5213](https://github.com/confluentinc/ksql/pull/5213)) ([0b3899a](https://github.com/confluentinc/ksql/commit/0b3899a4a31f668a897d82d98c59365aae8022ec))
+
+
+
+### Reverts
+
+* Revert "build: reference plugin URL as protperty (MINOR) (#5497)" ([093955f](https://github.com/confluentinc/ksql/commit/093955f9ce875a0b8656aaec78d4c619ebdc83ff)), closes [#5497](https://github.com/confluentinc/ksql/issues/5497)
```
Comment (big-andy-coates) [inline CHANGELOG.md:76]: remove? Internal change.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,136 @@
...
+* Revert "build ksqlDB against 6.0.0 (#5339)" (#5347) ([0a1bdad](https://github.com/confluentinc/ksql/commit/0a1bdadb7745efd2d3a5859ab5018719a2ed7f1e)), closes [#5339](https://github.com/confluentinc/ksql/issues/5339) [#5347](https://github.com/confluentinc/ksql/issues/5347)
```
Comment (big-andy-coates) [inline CHANGELOG.md:77]: remove? Internal change.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,136 @@
...
+* Any key name (#5093) ([1f0ca3e](https://github.com/confluentinc/ksql/commit/1f0ca3efb0f5ecadc7a604f219cbc16b2e120d8d)), closes [#5093](https://github.com/confluentinc/ksql/issues/5093)
```
Comment (big-andy-coates) [inline CHANGELOG.md:78]: This is not a revert. This is a feature... and a big one, even if I do say so myself ;)

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,136 @@
...
+* fail AVRO/Protobuf/JSON Schema statements if SR is missing ([#5597](https://github.com/confluentinc/ksql/pull/5597)) ([85a0320](https://github.com/confluentinc/ksql/commit/85a0320e774d236bda4898f82eb5d5c2f7b42cec))
+* fail on WINDOW clause without matching GROUP BY ([#5431](https://github.com/confluentinc/ksql/pull/5431)) ([68354d4](https://github.com/confluentinc/ksql/commit/68354d488aec07dc0bc9a7354ee8b8794bfef4e6))
+* improve print topic ([#5552](https://github.com/confluentinc/ksql/pull/5552)) ([e193576](https://github.com/confluentinc/ksql/commit/e193576c0f6c0295da6f4357391639e49a45863b))
+* Improve pull query error logging ([#5477](https://github.com/confluentinc/ksql/pull/5477)) ([f23412e](https://github.com/confluentinc/ksql/commit/f23412e9103531d2b8ce96cfcee3eff92784239b))
+* Java client error handling ([#5360](https://github.com/confluentinc/ksql/pull/5360)) ([4dca4dc](https://github.com/confluentinc/ksql/commit/4dca4dc4f56d9448bc0e66ea39cc728b2cf0a8fa))
```
Comment (vcrfxia) [inline CHANGELOG.md:53]: We can remove this since this is the first release with the Java client.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,136 @@
...
+* Remove stacktrace from error message ([#5478](https://github.com/confluentinc/ksql/pull/5478)) ([b63d7e8](https://github.com/confluentinc/ksql/commit/b63d7e8d3fb325a74120e40e3fc96d51e9c35856))
+* Retry on connection closed ([#5515](https://github.com/confluentinc/ksql/pull/5515)) ([8eb1f88](https://github.com/confluentinc/ksql/commit/8eb1f88c165aa0bb494288c46cd96ac911d7cfdf))
+* set retention.ms to -1 instead of Long.MAX_VALUE ([#5560](https://github.com/confluentinc/ksql/pull/5560)) ([22da8a0](https://github.com/confluentinc/ksql/commit/22da8a091f0020bbd541ad5f812054c7744f0785))
+* update Concat UDF to new framework and make variadic ([#5513](https://github.com/confluentinc/ksql/pull/5513)) ([cab6a86](https://github.com/confluentinc/ksql/commit/cab6a8672f97d86da75a3e5b1e70985146129640))
+* zero decimal bug ([#5531](https://github.com/confluentinc/ksql/pull/5531)) ([1b9094b](https://github.com/confluentinc/ksql/commit/1b9094bd5ead8a8441ef8f10a5f86722dd7d3988))
+* **client:** expose query ID from executeQuery() return type ([#5391](https://github.com/confluentinc/ksql/pull/5391)) ([bd3c642](https://github.com/confluentinc/ksql/commit/bd3c64289dc315fedd1f5f4040d571a5bf26323a))
```
Comment (vcrfxia) [inline CHANGELOG.md:65]: We can remove this since this is the first release with the Java client.

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,136 @@
...
+* **client:** fail queryID future if request fails (MINOR) ([#5435](https://github.com/confluentinc/ksql/pull/5435)) ([ddaf9f9](https://github.com/confluentinc/ksql/commit/ddaf9f998009cf58ee68133dac143191c0e1ad9a))
```
Comment (vcrfxia) [inline CHANGELOG.md:66]: We can remove this since this is the first release with the Java client.

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,136 @@
...
+
+* Revert "build: reference plugin URL as protperty (MINOR) (#5497)" ([093955f](https://github.com/confluentinc/ksql/commit/093955f9ce875a0b8656aaec78d4c619ebdc83ff)), closes [#5497](https://github.com/confluentinc/ksql/issues/5497)
+* Revert "build ksqlDB against 6.0.0 (#5339)" (#5347) ([0a1bdad](https://github.com/confluentinc/ksql/commit/0a1bdadb7745efd2d3a5859ab5018719a2ed7f1e)), closes [#5339](https://github.com/confluentinc/ksql/issues/5339) [#5347](https://github.com/confluentinc/ksql/issues/5347)
+* Any key name (#5093) ([1f0ca3e](https://github.com/confluentinc/ksql/commit/1f0ca3efb0f5ecadc7a604f219cbc16b2e120d8d)), closes [#5093](https://github.com/confluentinc/ksql/issues/5093)
+
+
+
+### BREAKING CHANGES
+
+* Statements containing PARTITION BY, GROUP BY or JOIN clauses will now produce different output schemas.
```
Comment (vcrfxia) [inline CHANGELOG.md:84]: Are all the lines below this one from the same breaking change? If so they need to be indented so the markdown formatting is correct.

We should probably update the changelog script to do this automatically. I'll look into that for the next release.

**Comment by derekjn** [comment]
Just a couple of high-level points on this changelog:
* Can we put all the new UDFs together in a list?
* Can we put each breaking change under its own subheading? The `PRIMARY KEY` change as written seems like it's a bit easy to miss
Other than that this looks good to me 👍

**Comment by stevenpyzhang** [comment]
Will do a follow up PR once 0.10.0 release is out.

**Pull request merged**